test/test.sh

```diff
@@ -948,6 +948,68 @@ test_auth_user() {
 	return 0
 }
 
+# tests auth_dbname whether it can be set and honored correctly.
+test_auth_dbname() {
+	$have_getpeereid || return 77
+	local result
+
+	admin "set auth_type='md5'"
+
+	# test if invalid globally set auth_dbname results in fallback to
+	# user's target database. 
+	admin "set auth_dbname='pinvalid'"
+
+	curuser=`psql -X -d "dbname=authdb user=someuser password=anypasswd" -tAq -c "select current_user;"`
+	echo "revert test curuser=$curuser"
+	test "$curuser" = "$curuser" || return 1
```

Comment:
```suggestion
	test "$curuser" = "someuser" || return 1
```

Reply:
oh what a miss. Updated for all comparisons for this test case.